lib/relation-definition.js

```diff
@@ -2113,34 +2113,54 @@ EmbedsOne.prototype.create = function(targetModelData, options, cb) {
 
   var inst = this.callScopeMethod('build', targetModelData);
 
-  var updateEmbedded = function() {
+  var updateEmbedded = function(callback) {
     if (modelInstance.isNewRecord()) {
       modelInstance.setAttribute(propertyName, inst);
       modelInstance.save(options, function(err) {
-        cb(err, err ? null : inst);
+        callback(err, err ? null : inst);
       });
     } else {
       modelInstance.updateAttribute(propertyName,
         inst, options, function(err) {
-          cb(err, err ? null : inst);
+          callback(err, err ? null : inst);
         });
     }
   };
 
   if (this.definition.options.persistent) {
     inst.save(options, function(err) { // will validate
       if (err) return cb(err, inst);
-      updateEmbedded();
+      updateEmbedded(cb);
     });
   } else {
-    var err = inst.isValid() ? null : new ValidationError(inst);
-    if (err) {
-      process.nextTick(function() {
-        cb(err);
-      });
-    } else {
-      updateEmbedded();
-    }
+    var context = {
+      Model: modelTo,
+      instance: inst,
+      options: options || {},
+      hookState: {},
+    };
+    modelTo.notifyObserversOf('before save', context, function(err) {
+      if (err) {
+        return process.nextTick(function() {
+          cb(err);
+        });
+      }
+
+      var err = inst.isValid() ? null : new ValidationError(inst);
+      if (err) {
+        process.nextTick(function() {
+          cb(err);
+        });
+      } else {
+        updateEmbedded(function(err, inst) {
+          if (err) return cb(err);
+          context.instance = inst;
```

Comment:
`ctx.isNewInstance` is marked as `// TODO`. Should I copy those TODO comments over, to make this more clear?